### PR TITLE
Optimize heartbeat

### PR DESCRIPTION
**Pull request by sherinkurian**
@nickelser this version is being battle-tested on sherin.clock.logistics.logistics

### CODE REVIEW

File: lib/zhong.rb [hunk 1]
```diff
@@ -41,8 +41,19 @@ def self.latest_heartbeat
   end
 
   def self.all_heartbeats
-    Zhong::Util.safe_mget(Zhong.redis.scan_each(match: "zhong:heartbeat:*").to_a).map do |k, v|
-      host, pid = k.split("zhong:heartbeat:", 2)[1].split("#", 2)
+    heartbeat_key = scheduler.config[:heartbeat_key]
+    heartbeats = Zhong.redis.hgetall(heartbeat_key)
+    heartbeats.select do |k, v|
+      if Time.at(v.to_i) > 15.minutes.ago
+        Zhong.redis.hdel(heartbeat_key, k)
```
Comment (nickelser) [inline lib/zhong.rb]: I believe you can `hdel` multiple keys at once: http://redis.io/commands/hdel, so I think a slight improvement would be to collect all expired keys, and do one `hdel` at the end, and return the unexpired ones.

**Comment by nickelser** [comment]
besides my one comment, if you wouldn't mind fixing the tests (`bundle exec rake` should run all the tests), this looks great!

**Comment by sherinkurian** [comment]
thanks for the feedback! fixed.

File: lib/zhong.rb [hunk 2]
```diff
@@ -41,8 +41,18 @@ def self.latest_heartbeat
   end
 
   def self.all_heartbeats
-    Zhong::Util.safe_mget(Zhong.redis.scan_each(match: "zhong:heartbeat:*").to_a).map do |k, v|
-      host, pid = k.split("zhong:heartbeat:", 2)[1].split("#", 2)
+    heartbeat_key = scheduler.config[:heartbeat_key]
+    heartbeats = Zhong.redis.hgetall(heartbeat_key)
+    old_beats, new_beats = heartbeats.partition do |k, v|
```
Comment (nickelser) [inline lib/zhong.rb]: nice! 👍 💯

**Pull request merged**